plugins/generator-1.15.2/forge-1.15.2/templates/block.java.ftl

```diff
@@ -148,7 +149,15 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
 
 		<#macro blockProterties>
 			Block.Properties.create(Material.${data.material})
-				.sound(SoundType.${data.soundOnStep})
+				<#if data.isCustomSoundType>
+					.sound(new SoundType(1.0f, 1.0f, new SoundEvent(new ResourceLocation("${data.breakSound}")),

```

Comment:
You need to use ForgeSoundType. More on why in docs:

![image](https://user-images.githubusercontent.com/16374228/119946707-34b24f80-bf97-11eb-9aa3-91dff178d81f.png)

Applies to both generators

Reply:
Yeah but then why did you not use `ForgeSoundType` here?
`.sound(SoundType.${data.soundOnStep})`

Reply:
Because it references existing sound, not new one.. You are making a new object instance here which is fundamentally different